.github/workflows/helm-publish-new-package-version.yml

```diff
@@ -0,0 +1,86 @@
+name: Publish helm package
+
+on:
+  workflow_call:
+    inputs:
+      next-version:
+        description: "Next final version to be release"
+        type: string
+        required: true
+      chart-dir:
+        description: "Path to the directory holding Chart.yml file"
+        type: string
+        required: false
+      helm-charts-repo:
+        description: "The name of the repository where the package will be added"
+        type: string
+        required: true
+      helm-charts-repo-branch:
+        description: "The name of the branch where the package will be added"
+        type: string
+        required: true
+      helm-charts-repo-subfolder:
+        description: "The name of the subfolder inside the charts repository where the package should be added"
+        type: string
+        required: false
+      helm-charts-repo-base-url:
+        description: "Helm chart repo base url"
+        type: string
+        required: false
+    outputs:
+      version:
+        description: "The version of the new published pacakage"
+        value: ${{ jobs.publish.outputs.version }}
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' }}
```

Comment:
A reusable workflow acts as if it was defined inside the caller. For instance, if a reusable workflow uses `actions/checkout`, it will checkout the repository of the caller, not the repository where the reusable workflow is defined. But that's a good point, it's probably better to move this restriction to the caller level, so it's more flexible. I'll do that.